deltacat/compute/compactor/model/compaction_session_audit_info.py

```diff
@@ -672,6 +683,20 @@ def set_output_size_bytes(
         self["outputSizeBytes"] = output_size_bytes
         return output_size_bytes
 
+    def set_output_record_count(
+        self, output_records: int
+    ) -> CompactionSessionAuditInfo:
+        if output_records < 0:
+            raise ValueError(
+                f"Output record count cannot be negative: {output_records}"
+            )
+        if not isinstance(output_records, int):
```

Comment:
We could allow `float` as well (and convert it to `int` before persisting) for robustness. Because, `int` can converted to `float` in an arithmetic operation unintentionally.